### ISSUE TITLE
After using HFTracer, the Bert model can't be trained

### ISSUE THREAD

**Issue by XGZhang11** [huggingface/transformers#16642]
## Environment info

- `transformers` version: 4.17.0
- Platform: ubuntu18.04
- Python version: 3.6.9
- PyTorch version (GPU?): 1.10.0 GPU
- Tensorflow version (GPU?): none
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

## Information

Model I am using (Bert, XLNet ...): Bert

The problem arises when using:
* [ ] the official example scripts: (give details below)
* my own modified scripts: (give details below) 

The tasks I am working on is:
* an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

1. I traced a BERT model using "HFTracer".  And i removed "labels" from "concrete_args" to use the "trainer.train".
2. After training of the modified model, i want to export an onnx model.
3. But "labels" has been fixed as an input of the modified model. And it is difficult to remove it. May i trace the bert model(or a part of it) without "labels", and then using the "trainer.train"?

## Expected behavior
May i trace the bert model(or a part of it) without "labels", and then using the "trainer.train"?
